### PR TITLE
Add forbid-duplicated-merge-keys option to key-duplicates rule

### LINKED ISSUE #561: allow warning on duplicate merge key
yamllint does not warn on duplicate merge keys:

```
main:
  <<: *anchor_one
  <<: *anchor_two
```

This should be done by merging a sequence like this:

    <<<: [*anchor_one, *anchor_two]

https://yaml.org/type/merge.html

This has a well defined ordering and you avoid differences due to parsers handling duplicates differently (basically for the same reason yamllint complains about duplicates in the first place).

Ignoring of duplicate merge keys was added intentionally https://github.com/adrienverge/yamllint/issues/88 so as its probably not uncommon to have this in files, maybe this warning should only be emitted when asked for.

### PR DESCRIPTION
**Pull request by travbale**
Closes #561

### CODE REVIEW

File: tests/rules/test_key_duplicates.py [hunk 1]
```diff
@@ -179,3 +179,44 @@ def test_key_tokens_in_flow_sequences(self):
...
+                   'Multiple Merge Keys are NOT OK:\n'
+                   'anchor_one: &anchor_one\n'
+                   '  one: one\n'
+                   'anchor_two: &anchor_two\n'
+                   '  two: two\n'
+                   'anchor_reference:\n'
+                   '  a: 1\n'
+                   '  <<: *anchor_one\n'
+                   '  b: 2\n'
+                   '  <<: *anchor_two\n', conf, problem=(11, 3))
```
Comment (adrienverge) [inline tests/rules/test_key_duplicates.py:204]: Can you add a case here to check when there are 3 `<<` merge markers? (I would expect 2 errors then)

File: yamllint/rules/key_duplicates.py [hunk 2]
```diff
@@ -16,6 +16,18 @@
 """
 Use this rule to prevent multiple entries with the same key in mappings.
 
+.. rubric:: Options
+
+* ``forbid-duplicated-merge-keys`` to forbid the usage of multiple merge keys.
```
Comment (adrienverge) [inline yamllint/rules/key_duplicates.py:21]: For consistency with other rules, I would propose:
```suggestion
* Use ``forbid-duplicated-merge-keys`` to forbid the usage of multiple merge
  keys.
```
or:
```suggestion
* Set ``forbid-duplicated-merge-keys`` to ``true`` to forbid the usage of
  multiple merge keys.
```

And also `merge keys.` → `merge keys ``<<``.`?

File: yamllint/rules/key_duplicates.py [hunk 3]
```diff
@@ -51,6 +63,29 @@
...
+   the following code snippet would **FAIL**:
+   ::
+
+    anchor_one: &anchor_one
+      one: one
+    anchor_two: &anchor_two
+      two: two
+    anchor_reference:
+      <<: *anchor_one
+      <<: *anchor_two]
```
Comment (adrienverge) [inline yamllint/rules/key_duplicates.py:88]: Small typo here:
```suggestion
      <<: *anchor_two
```

**Review by adrienverge — COMMENTED** [review]
Hello Travis, please excuse the delay to review this pull request.

Thanks for contributing and fixing #561, your commit is almost perfect, just see my 3 remarks.

**Comment by travbale** [comment]
Thanks for taking a look. Will work to get these in.

File: tests/rules/test_key_duplicates.py (hunk 1 above)
Comment (travbale) [reply tests/rules/test_key_duplicates.py:217]: Added

File: yamllint/rules/key_duplicates.py (hunk 3 above)
Comment (travbale) [reply yamllint/rules/key_duplicates.py]: Fixed

File: yamllint/rules/key_duplicates.py (hunk 2 above)
Comment (travbale) [reply yamllint/rules/key_duplicates.py]: Went with the first option.

**Comment by travbale** [comment]
Made the requested changes. I have an unrelated test failing I'm trying to figure out. 
```
======================================================================
FAIL: test_extended_ignore_str (tests.test_config.ExtendedConfigTestCase.test_extended_ignore_str)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travbale/yamllint/tests/test_config.py", line 388, in test_extended_ignore_str
    self.assertEqual(c.ignore.match_file('test.yaml'), False)
AssertionError: None != False
```

**Comment by adrienverge** [comment]
> Made the requested changes. I have an unrelated test failing I'm trying to figure out.
> 
> ```
> ======================================================================
> FAIL: test_extended_ignore_str (tests.test_config.ExtendedConfigTestCase.test_extended_ignore_str)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/Users/travbale/yamllint/tests/test_config.py", line 388, in test_extended_ignore_str
>     self.assertEqual(c.ignore.match_file('test.yaml'), False)
> AssertionError: None != False
> ```

Thanks a lot, I'll have a look!

Please ignore the failing test, it's due to a small bug in the new `pathspec` version, see https://github.com/cpburnz/python-pathspec/issues/84.

**Review by adrienverge — APPROVED** [review]
Thanks for contributing @travbale!

**Pull request merged**